From 2928468ef665e106339fcc1161a9c1e1ebcfc2a3 Mon Sep 17 00:00:00 2001
From: Dmitry Kalinkin <dmitry.kalinkin@gmail.com>
Date: Tue, 9 Oct 2018 13:57:30 +0200
Subject: [PATCH] ARROW-3467: [C++] Fix building against external
 double-conversion

Author: Dmitry Kalinkin <dmitry.kalinkin@gmail.com>

Closes #2728 from veprbl/pr/ARROW-3467 and squashes the following commits:

1ce23de8 <Dmitry Kalinkin> ARROW-3467: remove indirection by renaming external double-conversion target to double-conversion::double-conversion
595c62c1 <Dmitry Kalinkin> ARROW-3467: double-conversion 3.1.1+ uses double-conversion:: prefix for CMake target name
---
 cpp/CMakeLists.txt                          | 4 ++--
 cpp/cmake_modules/ThirdpartyToolchain.cmake | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/cpp/CMakeLists.txt b/cpp/CMakeLists.txt
index 4448b989e3..40575a7d8b 100644
--- a/cpp/CMakeLists.txt
+++ b/cpp/CMakeLists.txt
@@ -615,7 +615,7 @@ endif(UNIX)
 set(ARROW_LINK_LIBS)
 
 # Libraries to link statically with libarrow.so
-set(ARROW_STATIC_LINK_LIBS double-conversion)
+set(ARROW_STATIC_LINK_LIBS double-conversion::double-conversion)
 
 if (ARROW_WITH_BROTLI)
   SET(ARROW_STATIC_LINK_LIBS
@@ -694,7 +694,7 @@ else ()
   set(ARROW_MIN_TEST_LIBS
     arrow_shared
     ${ARROW_LINK_LIBS}
-    double-conversion
+    double-conversion::double-conversion
     ${BOOST_SYSTEM_LIBRARY}
     ${BOOST_FILESYSTEM_LIBRARY}
     ${BOOST_REGEX_LIBRARY}
diff --git a/cpp/cmake_modules/ThirdpartyToolchain.cmake b/cpp/cmake_modules/ThirdpartyToolchain.cmake
index e6c7f7828f..d8142293ba 100644
--- a/cpp/cmake_modules/ThirdpartyToolchain.cmake
+++ b/cpp/cmake_modules/ThirdpartyToolchain.cmake
@@ -489,10 +489,10 @@ else()
 endif()
 
 include_directories(SYSTEM ${DOUBLE_CONVERSION_INCLUDE_DIR})
-ADD_THIRDPARTY_LIB(double-conversion
-  STATIC_LIB ${DOUBLE_CONVERSION_STATIC_LIB})
 
 if (DOUBLE_CONVERSION_VENDORED)
+  ADD_THIRDPARTY_LIB(double-conversion::double-conversion
+    STATIC_LIB ${DOUBLE_CONVERSION_STATIC_LIB})
   add_dependencies(arrow_dependencies double-conversion_ep)
 endif()
 
